pkg/controllers/management/nodetemplate/nodetemplate_handler.go

```diff
@@ -0,0 +1,294 @@
+func (nt *nodeTemplateController) reviseNodePoolNodeTemplate(fullMigratedNTName, fullLegacyNTName string) error {
+	npList, err := nt.npLister.List("", labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, np := range npList {
+		if np.Spec.NodeTemplateName == fullLegacyNTName {
+			npCopy := np.DeepCopy()
+			npCopy.Spec.NodeTemplateName = fullMigratedNTName
+
+			if _, err := nt.npClient.Update(npCopy); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// reviseNodes searches for nodes that reference the old node template and replaces it with the new, global template
+func (nt *nodeTemplateController) reviseNodes(fullMigratedNTName, fullLegacyNTName string) error {
+	nodeList, err := nt.mgmtCtx.Management.Nodes("").Controller().Lister().List("", labels.Everything())
+	if err != nil {
+		return err
+	}
+	for _, node := range nodeList {
+		if node.Spec.NodeTemplateName == fullLegacyNTName {
+			nodeCopy := node.DeepCopy()
+			nodeCopy.Spec.NodeTemplateName = fullMigratedNTName
+
+			if _, err := nt.mgmtCtx.Management.Nodes("").Update(nodeCopy); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// getDynamicAnnotations returns the annotations of the given unstructured object returns by dynamic, k8s client
+func getDynamicAnnotations(dynamicNodeTemplate *unstructured.Unstructured, nodeTemplateName string) (map[string]interface{}, error) {
+	metadata, ok := dynamicNodeTemplate.Object["metadata"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("error fetching node template [%s] metadata", nodeTemplateName)
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+	if !ok {
+		annotations = make(map[string]interface{})
+	}
+
+	return annotations, nil
+}
+
+// writeDynamimcAnnotations adds the requested annotation to the given unstructured object returned by dynamic, k8s client
+func writeDynamimcAnnotations(dynamicNodeTemplate *unstructured.Unstructured, nodeTemplateName string, key string, value string) (*unstructured.Unstructured, error) {
+	metadata, ok := dynamicNodeTemplate.Object["metadata"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("error fetching node template [%s] metadata", nodeTemplateName)
+	}
+
+	annotations, ok := metadata["annotations"].(map[string]interface{})
+	if !ok {
+		annotations = make(map[string]interface{})
+	}
+
+	annotations[key] = value
+	metadata["annotations"] = annotations
+
+	dynamicNodeTemplate.Object["metadata"] = metadata
+
+	return dynamicNodeTemplate, nil
+}
+
+// createGlobalNodeTemplateClone returns the global clone of the given legacy node template. If one does not exist
+// it will be created
+func (nt *nodeTemplateController) createGlobalNodeTemplateClone(legacyName, cloneName string, dynamicNodeTemplate *unstructured.Unstructured,
+	client dynamic.NamespaceableResourceInterface) error {
+	if _, err := nt.nsLister.Get("", namespace.NodeTemplateGlobalNamespace); err != nil {
```

Comment:
Drop the ns create, if this doesn't already exist something bad happened in startup